packages/core/src/integrations/functiontostring.ts

```diff
@@ -22,10 +22,14 @@ export class FunctionToString implements Integration {
     // eslint-disable-next-line @typescript-eslint/unbound-method
     originalFunctionToString = Function.prototype.toString;
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    Function.prototype.toString = function (this: WrappedFunction, ...args: any[]): string {
-      const context = getOriginalFunction(this) || this;
-      return originalFunctionToString.apply(context, args);
-    };
+    // intrinsics (like Function.prototype) might be immutable in some environments
+    // e.g. Node with --frozen-intrinsics, XS (an embedded JavaScript engine) or SES (a JavaScript proposal)
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      Function.prototype.toString = function (this: WrappedFunction, ...args: any[]): string {
+        const context = getOriginalFunction(this) || this;
+        return originalFunctionToString.apply(context, args);
+      };
+    } catch {}
```

Comment:
```suggestion
    } catch {
      // ignore errors here, just don't patch this
    }
```

Need to make the block non-empty to make eslint happy!